### PR TITLE
Added an option to remove fingerprint

### PR DESCRIPTION
**Pull request by aug-riedinger**
As from the issue I opened: https://github.com/adunkman/connect-assets/issues/290
I added an option `fingerprints`, defaulted to `true` in prod, `false` in dev, acting the Sprocket way.

I added some tests, though I never managed to run them because `npm install --dev` fails due to some circular dependency issue, using `npm 2.2.0` and `node 0.10.25`. Here's my console output:

```
$ npm install --dev 
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@1.21.4' from dependencies
npm WARN package.json Dependency 'blanket' exists in both dependencies and devDependencies, using 'blanket@1.1.6' from dependencies
npm WARN package.json Dependency 'expect.js' exists in both dependencies and devDependencies, using 'expect.js@0.3.1' from dependencies
npm WARN package.json Dependency 'travis-cov' exists in both dependencies and devDependencies, using 'travis-cov@0.2.5' from dependencies
npm WARN package.json Dependency 'connect' exists in both dependencies and devDependencies, using 'connect@3.2.0' from dependencies
npm WARN package.json Dependency 'ejs' exists in both dependencies and devDependencies, using 'ejs@1.0.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@*' from dependencies
npm WARN package.json Dependency 'serve' exists in both dependencies and devDependencies, using 'serve@*' from dependencies
npm WARN package.json Dependency 'istanbul' exists in both dependencies and devDependencies, using 'istanbul@0.3.2' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@~1.21.4' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@~4.0.0' from dependencies
npm WARN package.json Dependency 'supertest' exists in both dependencies and devDependencies, using 'supertest@~0.13.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@*' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@*' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@*' from dependencies
npm WARN package.json Dependency 'nodeunit' exists in both dependencies and devDependencies, using 'nodeunit@^0.9.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@1.x.x' from dependencies
npm WARN package.json Dependency 'chai' exists in both dependencies and devDependencies, using 'chai@1.x.x' from dependencies
npm WARN package.json Dependency 'istanbul' exists in both dependencies and devDependencies, using 'istanbul@0.3.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@~1.21.4' from dependencies
npm WARN package.json Dependency 'readable-stream' exists in both dependencies and devDependencies, using 'readable-stream@~1.0.27' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@~4.0.1' from dependencies
npm WARN package.json Dependency 'supertest' exists in both dependencies and devDependencies, using 'supertest@~0.13.0' from dependencies
npm WARN package.json Dependency 'benchmark' exists in both dependencies and devDependencies, using 'benchmark@1.0.0' from dependencies
npm WARN package.json Dependency 'beautify-benchmark' exists in both dependencies and devDependencies, using 'beautify-benchmark@0.2.4' from dependencies
npm WARN package.json Dependency 'fast-url-parser' exists in both dependencies and devDependencies, using 'fast-url-parser@~1.0.0' from dependencies
npm WARN package.json Dependency 'istanbul' exists in both dependencies and devDependencies, using 'istanbul@0.3.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@~1.21.4' from dependencies
npm WARN package.json Dependency 'browserify' exists in both dependencies and devDependencies, using 'browserify@5.11.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@*' from dependencies
npm WARN package.json Dependency 'express' exists in both dependencies and devDependencies, using 'express@3.1.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@1.19.0' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@3.3.1' from dependencies
npm WARN package.json Dependency 'coffee-script' exists in both dependencies and devDependencies, using 'coffee-script@~1.7.1' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@~4.0.0' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@>= 0.0.1' from dependencies
npm WARN package.json Dependency 'phantomjs' exists in both dependencies and devDependencies, using 'phantomjs@0.2.2' from dependencies
npm WARN package.json Dependency 'tap' exists in both dependencies and devDependencies, using 'tap@~0.2.6' from dependencies
npm WARN package.json Dependency 'browserify' exists in both dependencies and devDependencies, using 'browserify@latest' from dependencies
npm WARN package.json Dependency 'mocha-better-spec-reporter' exists in both dependencies and devDependencies, using 'mocha-better-spec-reporter@latest' from dependencies
npm WARN package.json Dependency 'gulp-uglify' exists in both dependencies and devDependencies, using 'gulp-uglify@^1.0.1' from dependencies
npm WARN package.json Dependency 'gulp-util' exists in both dependencies and devDependencies, using 'gulp-util@^2.2.14' from dependencies
npm WARN package.json Dependency 'gulp' exists in both dependencies and devDependencies, using 'gulp@^3.8.10' from dependencies
npm WARN package.json Dependency 'vinyl-source-stream2' exists in both dependencies and devDependencies, using 'vinyl-source-stream2@^0.1.1' from dependencies
npm WARN package.json Dependency 'gulp-load-plugins' exists in both dependencies and devDependencies, using 'gulp-load-plugins@^0.5.1' from dependencies
npm WARN package.json Dependency 'gulp-rename' exists in both dependencies and devDependencies, using 'gulp-rename@^1.2.0' from dependencies
npm WARN package.json Dependency 'gulp-header' exists in both dependencies and devDependencies, using 'gulp-header@^1.2.2' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@latest' from dependencies
npm WARN package.json Dependency 'zuul' exists in both dependencies and devDependencies, using 'zuul@latest' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@*' from dependencies
npm WARN package.json Dependency 'expect.js' exists in both dependencies and devDependencies, using 'expect.js@*' from dependencies
npm WARN package.json Dependency 'serve' exists in both dependencies and devDependencies, using 'serve@*' from dependencies
npm WARN package.json Dependency 'assetgraph' exists in both dependencies and devDependencies, using 'assetgraph@=1.0.1' from dependencies
npm WARN package.json Dependency 'assetgraph-builder' exists in both dependencies and devDependencies, using 'assetgraph-builder@=1.0.3' from dependencies
npm WARN package.json Dependency 'jamjs' exists in both dependencies and devDependencies, using 'jamjs@=0.2.17' from dependencies
npm WARN package.json Dependency 'optimist' exists in both dependencies and devDependencies, using 'optimist@=0.3.1' from dependencies
npm WARN package.json Dependency 'uglify-js' exists in both dependencies and devDependencies, using 'uglify-js@=2.4.15' from dependencies
npm WARN package.json Dependency 'vows' exists in both dependencies and devDependencies, using 'vows@=0.7.0' from dependencies
npm WARN package.json Dependency 'tape' exists in both dependencies and devDependencies, using 'tape@~1.0.4' from dependencies
npm WARN package.json Dependency 'tap' exists in both dependencies and devDependencies, using 'tap@~0.4.0' from dependencies
npm WARN package.json Dependency 'covert' exists in both dependencies and devDependencies, using 'covert@^1.0.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@~1.17.1' from dependencies
npm WARN package.json Dependency 'supertest' exists in both dependencies and devDependencies, using 'supertest@~0.9.0' from dependencies
npm WARN package.json Dependency 'express' exists in both dependencies and devDependencies, using 'express@~3.4.7' from dependencies
npm WARN package.json Dependency 'fs-extra' exists in both dependencies and devDependencies, using 'fs-extra@~0.11.0' from dependencies
npm WARN package.json Dependency 'benchmark' exists in both dependencies and devDependencies, using 'benchmark@*' from dependencies
npm WARN package.json Dependency 'uglify-js' exists in both dependencies and devDependencies, using 'uglify-js@~2.4.0' from dependencies
npm WARN package.json Dependency 'source-map' exists in both dependencies and devDependencies, using 'source-map@~0.1.27' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@1.17.x' from dependencies
npm WARN package.json Dependency 'gulp-es6-transpiler' exists in both dependencies and devDependencies, using 'gulp-es6-transpiler@1.0.0' from dependencies
npm WARN package.json Dependency 'gulp-json-editor' exists in both dependencies and devDependencies, using 'gulp-json-editor@2.0.2' from dependencies
npm WARN package.json Dependency 'jshint-stylish' exists in both dependencies and devDependencies, using 'jshint-stylish@1.0.0' from dependencies
npm WARN package.json Dependency 'gulp-jshint' exists in both dependencies and devDependencies, using 'gulp-jshint@1.8.4' from dependencies
npm WARN package.json Dependency 'gonzales-pe' exists in both dependencies and devDependencies, using 'gonzales-pe@3.0.0-10' from dependencies
npm WARN package.json Dependency 'gulp-mocha' exists in both dependencies and devDependencies, using 'gulp-mocha@1.1.0' from dependencies
npm WARN package.json Dependency 'gulp-util' exists in both dependencies and devDependencies, using 'gulp-util@3.0.1' from dependencies
npm WARN package.json Dependency 'fs-extra' exists in both dependencies and devDependencies, using 'fs-extra@0.12.0' from dependencies
npm WARN package.json Dependency 'gonzales' exists in both dependencies and devDependencies, using 'gonzales@1.0.7' from dependencies
npm WARN package.json Dependency 'request' exists in both dependencies and devDependencies, using 'request@2.44.0' from dependencies
npm WARN package.json Dependency 'rework' exists in both dependencies and devDependencies, using 'rework@1.0.1' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@4.0.4' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@1.21.4' from dependencies
npm WARN package.json Dependency 'cssom' exists in both dependencies and devDependencies, using 'cssom@0.3.0' from dependencies
npm WARN package.json Dependency 'gulp' exists in both dependencies and devDependencies, using 'gulp@3.8.8' from dependencies
npm WARN package.json Dependency 'es6-transpiler' exists in both dependencies and devDependencies, using 'es6-transpiler@0.7.16' from dependencies
npm WARN package.json Dependency 'grunt' exists in both dependencies and devDependencies, using 'grunt@~0.4.1' from dependencies
npm WARN package.json Dependency 'grunt-contrib-jshint' exists in both dependencies and devDependencies, using 'grunt-contrib-jshint@~0.6.4' from dependencies
npm WARN package.json Dependency 'jshint-stylish' exists in both dependencies and devDependencies, using 'jshint-stylish@latest' from dependencies
npm WARN package.json Dependency 'acorn' exists in both dependencies and devDependencies, using 'acorn@~0.3.1' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@~1.12.1' from dependencies
npm WARN package.json Dependency 'grunt-cli' exists in both dependencies and devDependencies, using 'grunt-cli@~0.1.9' from dependencies
npm WARN package.json Dependency 'bluebird' exists in both dependencies and devDependencies, using 'bluebird@~0.11' from dependencies
npm WARN package.json Dependency 'benchmark' exists in both dependencies and devDependencies, using 'benchmark@~1.0.0' from dependencies
npm WARN package.json Dependency 'deque' exists in both dependencies and devDependencies, using 'deque@0.0.4' from dependencies
npm WARN package.json Dependency 'q' exists in both dependencies and devDependencies, using 'q@~0.9.7' from dependencies
npm WARN package.json Dependency 'semver-utils' exists in both dependencies and devDependencies, using 'semver-utils@~1.1.0' from dependencies
npm WARN package.json Dependency 'tap' exists in both dependencies and devDependencies, using 'tap@*' from dependencies
npm WARN package.json Dependency 'nodeunit' exists in both dependencies and devDependencies, using 'nodeunit@~0.7.4' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@~1.6' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@~1.2' from dependencies
npm WARN package.json Dependency 'tap' exists in both dependencies and devDependencies, using 'tap@~0.4.8' from dependencies
npm WARN package.json Dependency 'browserify' exists in both dependencies and devDependencies, using 'browserify@latest' from dependencies
npm WARN package.json Dependency 'mocha-better-spec-reporter' exists in both dependencies and devDependencies, using 'mocha-better-spec-reporter@0.0.2' from dependencies
npm WARN package.json Dependency 'gulp-mocha' exists in both dependencies and devDependencies, using 'gulp-mocha@^0.4.1' from dependencies
npm WARN package.json Dependency 'gulp-uglify' exists in both dependencies and devDependencies, using 'gulp-uglify@^0.3.0' from dependencies
npm WARN package.json Dependency 'gulp-util' exists in both dependencies and devDependencies, using 'gulp-util@^2.2.14' from dependencies
npm WARN package.json Dependency 'gulp' exists in both dependencies and devDependencies, using 'gulp@^3.6.2' from dependencies
npm WARN package.json Dependency 'vinyl-source-stream2' exists in both dependencies and devDependencies, using 'vinyl-source-stream2@^0.1.1' from dependencies
npm WARN package.json Dependency 'gulp-load-plugins' exists in both dependencies and devDependencies, using 'gulp-load-plugins@^0.5.1' from dependencies
npm WARN package.json Dependency 'gulp-rename' exists in both dependencies and devDependencies, using 'gulp-rename@^1.2.0' from dependencies
npm WARN package.json Dependency 'gulp-header' exists in both dependencies and devDependencies, using 'gulp-header@^1.0.2' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@^1.19.0' from dependencies
npm WARN package.json Dependency 'zuul' exists in both dependencies and devDependencies, using 'zuul@^1.6.5' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@>= 0.0.1' from dependencies
npm WARN package.json Dependency 'tape' exists in both dependencies and devDependencies, using 'tape@~2.12.3' from dependencies
npm WARN package.json Dependency 'vinyl-map' exists in both dependencies and devDependencies, using 'vinyl-map@1.0.1' from dependencies
npm WARN package.json Dependency 'gulp-rename' exists in both dependencies and devDependencies, using 'gulp-rename@~1.2.0' from dependencies
npm WARN package.json Dependency 'gulp' exists in both dependencies and devDependencies, using 'gulp@~3.6.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@^2.0.0' from dependencies
npm WARN package.json Dependency 'mocha-better-spec-reporter' exists in both dependencies and devDependencies, using 'mocha-better-spec-reporter@latest' from dependencies
npm WARN package.json Dependency 'nodeunit' exists in both dependencies and devDependencies, using 'nodeunit@>0.0.0' from dependencies
npm WARN package.json Dependency 'uglify-js' exists in both dependencies and devDependencies, using 'uglify-js@1.2.x' from dependencies
npm WARN package.json Dependency 'nodelint' exists in both dependencies and devDependencies, using 'nodelint@>0.0.0' from dependencies
npm WARN package.json Dependency 'browserify' exists in both dependencies and devDependencies, using 'browserify@6.1.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@*' from dependencies
npm WARN package.json Dependency 'event-stream' exists in both dependencies and devDependencies, using 'event-stream@^3.1.7' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@*' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@*' from dependencies
npm WARN package.json Dependency 'vinyl' exists in both dependencies and devDependencies, using 'vinyl@*' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@>= 0.0.1' from dependencies
npm WARN package.json Dependency 'jshint' exists in both dependencies and devDependencies, using 'jshint@^2.5.1' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@^1.18.2' from dependencies
npm WARN package.json Dependency 'proxyquire' exists in both dependencies and devDependencies, using 'proxyquire@^1.0.1' from dependencies
npm WARN package.json Dependency 'sinon' exists in both dependencies and devDependencies, using 'sinon@^1.9.1' from dependencies
npm WARN package.json Dependency 'coveralls' exists in both dependencies and devDependencies, using 'coveralls@^2.7.0' from dependencies
npm WARN package.json Dependency 'graceful-fs' exists in both dependencies and devDependencies, using 'graceful-fs@^3.0.0' from dependencies
npm WARN package.json Dependency 'istanbul' exists in both dependencies and devDependencies, using 'istanbul@^0.3.0' from dependencies
npm WARN package.json Dependency 'jshint' exists in both dependencies and devDependencies, using 'jshint@^2.5.0' from dependencies
npm WARN package.json Dependency 'jshint-stylish' exists in both dependencies and devDependencies, using 'jshint-stylish@^1.0.0' from dependencies
npm WARN package.json Dependency 'marked-man' exists in both dependencies and devDependencies, using 'marked-man@^0.1.3' from dependencies
npm WARN package.json Dependency 'mkdirp' exists in both dependencies and devDependencies, using 'mkdirp@^0.5.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@^2.0.1' from dependencies
npm WARN package.json Dependency 'mocha-lcov-reporter' exists in both dependencies and devDependencies, using 'mocha-lcov-reporter@^0.0.1' from dependencies
npm WARN package.json Dependency 'q' exists in both dependencies and devDependencies, using 'q@^1.0.0' from dependencies
npm WARN package.json Dependency 'rimraf' exists in both dependencies and devDependencies, using 'rimraf@^2.2.5' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@^4.0.0' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@latest' from dependencies
npm WARN package.json Dependency 'tap' exists in both dependencies and devDependencies, using 'tap@~0.4.0' from dependencies
npm WARN package.json Dependency 'map-stream' exists in both dependencies and devDependencies, using 'map-stream@>=0.0.4' from dependencies
npm WARN package.json Dependency 'mocha' exists in both dependencies and devDependencies, using 'mocha@>=1.15.0' from dependencies
npm WARN package.json Dependency 'should' exists in both dependencies and devDependencies, using 'should@>=2.1.0' from dependencies
npm WARN package.json Dependency 'gulp' exists in both dependencies and devDependencies, using 'gulp@>=3.0.0' from dependencies
npm WARN package.json Dependency 'gulp-jshint' exists in both dependencies and devDependencies, using 'gulp-jshint@>=1.1.0' from dependencies
npm WARN package.json Dependency 'argg' exists in both dependencies and devDependencies, using 'argg@0.0.1' from dependencies
npm WARN package.json Dependency 'codeclimate-test-reporter' exists in both dependencies and devDependencies, using 'codeclimate-test-reporter@0.0.3' from dependencies
npm WARN package.json Dependency 'gulp-concat' exists in both dependencies and devDependencies, using 'gulp-concat@>=2.3.4 <3.0.0-0' from dependencies
npm WARN package.json Dependency 'gulp-sourcemaps' exists in both dependencies and devDependencies, using 'gulp-sourcemaps@>=1.1.1 <2.0.0-0' from dependencies
npm WARN package.json Dependency 'istanbul' exists in both dependencies and devDependencies, using 'istanbul@>=0.3.0 <0.4.0-0' from dependencies
npm WARN package.json Dependency 'rimraf' exists in both dependencies and devDependencies, using 'rimraf@>=2.2.8 <3.0.0-0' from dependencies
npm WARN package.json Dependency 'tape' exists in both dependencies and devDependencies, using 'tape@>=2.12.3 <3.0.0-0' from dependencies
npm WARN package.json Dependency 'vinyl' exists in both dependencies and devDependencies, using 'vinyl@>=0.3.2 <0.4.0-0' from dependencies
```

I am willing to try to fix that but I don't know how yet.

### CODE REVIEW

**Comment by adunkman** [comment]
Okay!

I don’t like the Sprockets approach, since you can shoot yourself in the foot if you don’t know what you’re doing — you can hardcode paths in development to your assets, but when you release in production suddenly all your paths change (they get the fingerprints appended) and all your assets are broken. 

However, I totally get that it’s a pain in the ass to use developer tools like the web inspector with JavaScript, since things like your file breakpoints are dependent on the filename (so, every time you make a change your breakpoints are lost). 

I tried to move the fingerprints to a query string parameter so we can keep the “you must use the asset url helpers” behavior — but that doesn’t fix anything; breakpoints are still lost in the web inspector.

File: index.js [hunk 1]
```diff
@@ -65,6 +65,7 @@ var parseOptions = module.exports._parseOptions = function (options) {
   options.compile = options.compile != null ? options.compile : true;
   options.compress = options.compress != null ? options.compress : isProduction;
   options.gzip = options.gzip != null ? options.gzip : false;
+  options.fingerprints = options.fingerprints || isProduction ? true : false
```
Comment (adunkman) [inline index.js]: This isn’t actually correct — I’ll make the change when merging though. With this line, you cannot disable fingerprints in production. If you follow the pattern of the lines above, you’re able to override the value regardless of environment.

**Comment by adunkman** [comment]
This change introduces some odd behavior when compiling assets out to disk — since the `manifest.json` file uses fingerprints, the files stored to disk are stored using fingerprints. If fingerprinting isn’t enabled when `compile: true`, then the compiled assets that reference each other (say, via CSS `@import`) reference the filenames without fingerprints (which don’t exist on disk).

I’m not sure it’s an issue. It’s a bit strange, but people will complain if it’s wrong!

**Comment by adunkman** [comment]
Okay — made a few changes in 60c2b3777d09a9de3b96e8fdb3c9efd9a9d32f8f. Published as `v4.5.0`.

**Comment by aug-riedinger** [comment]
:+1: Super nice!

By the way, I recently realized that in Rails 4, Sprocket does add the fingerprint in development environment. I could not find anything about the why this change occured. But I still think it is pretty usefull to have it as an option (for developper tool integration mostly indeed).
So If you'd rather be in line with Sprocket, I guess the default option value should be `true` even in development mode, but still keeping it as an option!

**Pull request merged**